plugins/inputs/consul/consul.go

```diff
@@ -141,6 +146,14 @@ func (c *Consul) GatherHealthCheck(acc telegraf.Accumulator, checks []*api.Healt
 	}
 }
 
+func isValidTagName(tag string) bool {
+	matched, err := regexp.MatchString(`^[a-zA-Z_][a-zA-Z0-9_]*`, tag)
```

Comment:
Compile the regex, either at package scope or once at plugin startup.  Otherwise this will compile the regex on every item.